File: src/ts/terrain/landscape/constantLayers.ts
import { simpleElevationFunction } from "./elevationFunction";

export function constantLayer(value: number): simpleElevationFunction {
    return () => value;
}

export function zeroLayer(): simpleElevationFunction {
    return constantLayer(0);
}

export function oneLayer(): simpleElevationFunction {
    return constantLayer(1);
}

File: packages/terrain-generation/src/landscape/constant_layers.rs
use crate::utils::vector3::Vector3;

pub fn constant_layer(value: f32) -> impl Fn(&Vector3, f32, &mut Vector3) -> f32 {
    move |_: &Vector3, _: f32, _: &mut Vector3| value
}

pub fn zero_layer() -> impl Fn(&Vector3, f32, &mut Vector3) -> f32 {
    constant_layer(0.0)
}

pub fn one_layer() -> impl Fn(&Vector3, f32, &mut Vector3) -> f32 {
    constant_layer(1.0)
}
